bcgov/nr-compliance-enforcement
# fix: CE-1198-Cannot add animal after emptied drugs

Addresses bug [CE-1198](https://env-ceds.atlassian.net/browse/CE-1198). A type error was flagging when drugs were added, then removed, from an animal outcome. Additional refactoring is also included in this. 

## Changes 🚧

- Modified save validation with optional chaining to ensure undefined rather than error.
- Refactored `getValue` ➡️ `getDropdownOption` used for dropdown options to operate as a common method. 

## How Has This Been Tested? 🔬

- [x] Manual testing, matching from issue report.

### To test 🔬

1. In a complaint outcome, add an _outcome by animal_
2. Add a drug to the outcome, then delete it. 
3. Save the outcome. If there are validation errors, you should get a toast saying as such, not a full-page error. If there are no validation errors, it should save.

### Checklist 📃


[CE-1198]: https://env-ceds.atlassian.net/browse/CE-1198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-772-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-772-frontend.apps.silver.devops.gov.bc.ca/api)


Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)
